### DISCUSSION TITLE
[Art competition] Abstract Semantics

### DISCUSSION THREAD

**Discussion by irfanariq** [prosyslab-classroom/cs524-program-analysis#275] [Art competition]
![DALL·E 2022-11-29 21 02 23 -  a person creating an virtual character of himself_herself on social media](https://user-images.githubusercontent.com/22218209/204531590-25da773f-b9f6-4945-8e46-246061eca2d0.png)

## Short Description
For me, [designing abstract semantics] means [a person creating an avatar of himself/herself on social media]

## Long Description (up to 3 sentences)
someone may hide his/her bad side in the avatar he/she made but, the avatar should still represent the person in real life so that his/her friend on social media can recognize him/her in real life and do not disappointed when meeting him/her in real life. it is similar to designing abstract semantics from concrete semantics. we may hide some details of concrete semantics in abstract semantics but the abstract semantics should still represent the concrete semantics so that the program analysis has precise results and does not produce false alarms.

**Comment by KihongHeo** [comment]
That is interesting.

**Comment by KihongHeo** [comment]
What do you mean by 
> the abstract semantics should still represent the concrete semantics so that the program analysis has precise results and does not produce false alarms.

**Comment by irfanariq** [comment]
I mean better abstract semantics should represent concrete semantics more. For example, if we compare the sign domain and the interval domain. The Interval domain is better since it has more details of integer (in the concrete domain) compared to the sign domain. Thus, the program analysis result that uses the interval domain produces more precise results than the program analysis that uses the sign domain.
